src/main/webapp/script.js

```diff
@@ -11,3 +11,79 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
+
+$(document).ready(init);
+
+function init() {
+  loadTasks();
+}
+
+function loadTasks() {
+  fetchData("/load_tasks")
+      .then((tasks) => renderTasks(tasks))
+      .catch((status)=>console.error(status));
+}
+
+function renderTasks(tasks) {
+  $("task-list").empty();
+  tasks.forEach(renderSingleTask);
+}
+
+function renderSingleTask(task) {
+  id = task.id;
+  newTask = $("<p></p>", {"id": id});
+  checkBox = $("<input>", {"name": id, "type": "checkbox"});
+  checkBox.appendTo(newTask);
+  newTask.append(task.title);
+  newTask.addClass("task");
+  newTask.appendTo("#task-list");
+}
+
+/**
+ * Provides feedback to the user that tasks were scheduled, and
+ * reports if there are problems.
+ */
+function updateView(results) {
+  // TODO: show how many tasks were scheduled, notify about conflicts/problems
+}
+
+function schedule() {
+  const formContent = new FormData($("#task-list")[0]);
+  postData("/schedule", new URLSearchParams(formContent).toString())
+      .then((results) => updateView(results)); //This is still returning a promise
+}
+
+function fetchData(url) {
+  return fetch(url)
+      .then((response) => getJsonIfOk(response));
+}
+
+
+function postData(url, data) {
+  return fetch(url, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/x-www-form-urlencoded"
+        },
+        body: data
+      }
+  )
+      .then((response) => getJsonIfOk(response))
+      .catch((error) => handleFetchError(error));
```

Comment:
`handleFetchError` is no longer defined, probably it should be `handleNetworkError`